docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,314 @@
+		 <code>KSTREAM|KTABLE-&gt;operator name&lt;-&gt;number suffix&lt;</code>. The number is a
+		 globally incrementing number that represents the operator's order in the topology.
+		 The generated number is prefixed with a varying number of "0"s to create a
+		 string that is consistently 10 characters long.
+		 This means that if you add/remove or shift the order of operations, the position of the
+		 processor shifts, which shifts the name of the processor.  Since <strong>most</strong> processors exist
+		 in memory only, this name shifting on topology changes presents no issue.  But the name
+		 shifting does have implications for topologies with stateful operators or repartition topics. 
+
+		 Here's a different topology with some state:
+		 <code>
+		  builder.&gt;String,String&lt;stream("input")
+			     .groupByKey()
+			     .count()
+			     .toStream()
+			     .to("output");
+		 </code>
+
+		 This topology description yields the following:
+		 <code>
+		     Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-AGGREGATE-0000000002
+		    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000001])
+		      --> KTABLE-TOSTREAM-0000000003
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KTABLE-TOSTREAM-0000000003 (stores: [])
+		      --> KSTREAM-SINK-0000000004
+		      <-- KSTREAM-AGGREGATE-0000000002
+		    Sink: KSTREAM-SINK-0000000004 (topic: output)
+		      <-- KTABLE-TOSTREAM-0000000003
+
+		  </code>
+		  
+		  You can see from the topology description above that the state store is named
+		  <code>KSTREAM-AGGREGATE-STATE-STORE-0000000002</code>.  Here's what happens when you
+		  add a filter to keep some of the records out of the aggregation:
+		  <code>
+		    builder.&gt;String, String&lt;stream("input")
+		        .filter((k,v)-> v !=null && v.length() >= 6 )
+		        .groupByKey()
+		        .count()
+		        .toStream()
+		        .to("output"); 
+		  </code>  
+
+		  And the corresponding topology:
+		  <code>
+		  Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-FILTER-0000000001
+		    Processor: KSTREAM-FILTER-0000000001 (stores: [])
+		      --> KSTREAM-AGGREGATE-0000000003
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KSTREAM-AGGREGATE-0000000003 (stores: [KSTREAM-AGGREGATE-STATE-STORE-0000000002])
+		      --> KTABLE-TOSTREAM-0000000004
+		      <-- KSTREAM-FILTER-0000000001
+		    Processor: KTABLE-TOSTREAM-0000000004 (stores: [])
+		      --> KSTREAM-SINK-0000000005
+		      <-- KSTREAM-AGGREGATE-0000000003
+		    Sink: KSTREAM-SINK-0000000005 (topic: output)
+		      <-- KTABLE-TOSTREAM-0000000004
+		  </code>
+
+		Notice that since you've added an operation <em>before</em> the <code>count</code> operation, the state
+		  store (and the changelog topic as well) names have changed.  This name change means you can't
```

Comment:
prop: sentence reads a bit awkward, how about `...the state store (and changelog topic) names...` or `...the state store names (and changelog topics as well) have changed` ?

Reply:
ack